lib/engine/part/city.rb

```diff
@@ -110,6 +110,28 @@ def place_token(corporation, token, free: false, check_tokenable: true, cheater:
         if check_tokenable && !tokenable?(
             corporation, free: free, tokens: token, cheater: cheater, extra_slot: extra_slot, spender: spender
           )
+          if tile.cities.any? do |c|
+               c.tokened_by?(token.corporation)
+             end
+            raise GameError,
+                  "#{corporation.name} cannot lay token - already has a token on #{tile.hex&.id}"
+          end
+
+          if tile.token_blocked_by_reservation?(corporation)
+            raise GameError,
+                  "#{corporation.name} cannot lay token - remaining token slots are reserved on #{tile.hex&.id}"
+          end
+          if !free && token.price > (spender || corporation).cash
+            raise GameError,
+                  "#{corporation.name} cannot lay token - cannot afford token on #{tile.hex&.id}"
+          end
+          unless extra_slot || get_slot(
+token.corporation, cheater: cheater
+)
+            raise GameError,
+                  "#{corporation.name} cannot lay token - no token slots available on #{tile.hex&.id}"
+          end
+
```

Comment:
Since you mentioned DRY in slack, here is an idea of how to do that:

Create a method that returns the error string or reason (Ex. :reserved) if tokenable? fails and then have the various methods that make up tokenable? set that error string/reason?

I'm not entirely sure if this is a good idea or not, but I do think it is worth avoiding duplication and dependency between tokenable? and place_token.